modules/RTC/ortc/ortcUtils.js

```diff
@@ -0,0 +1,159 @@
+/* global __filename, RTCRtpReceiver */
+
+/**
+ * Get local ORTC RTP capabilities filtered and adapted to the given remote RTP
+ * capabilities.
+ * @param {RTCRtpCapabilities} filterWithCapabilities - RTP capabilities to
+ * filter with.
+ * @return {RTCRtpCapabilities}
+ */
+export function getLocalCapabilities(filterWithCapabilities) {
+    const localFullCapabilities = RTCRtpReceiver.getCapabilities();
+    const localCapabilities = {
+        codecs: [],
+        headerExtensions: [],
+        fecMechanisms: []
+    };
+
+    // Map of RTX and codec payloads.
+    // - index: Codec payloadType
+    // - value: Associated RTX payloadType
+    // @type {map<Number, Number>}
+    const remoteRtxMap = new Map();
+
+    // Set codecs.
+    for (const remoteCodec of filterWithCapabilities.codecs) {
+        const remoteCodecName = remoteCodec.name.toLowerCase();
+
+        if (remoteCodecName === 'rtx') {
+            remoteRtxMap.set(
+                remoteCodec.parameters.apt, remoteCodec.preferredPayloadType);
+
+            continue; // eslint-disable-line no-continue
+        }
+
+        for (const localCodec of localFullCapabilities.codecs) {
+            const localCodecName = localCodec.name.toLowerCase();
+
+            if (localCodecName !== remoteCodecName
+                || localCodec.kind !== remoteCodec.kind
+                || localCodec.clockRate !== remoteCodec.clockRate) {
+                continue; // eslint-disable-line no-continue
+            }
+
+            const codec = {
+                name: localCodec.name,
+                mimeType: `${localCodec.kind}/${localCodec.name}`,
+                kind: localCodec.kind,
+                preferredPayloadType: remoteCodec.preferredPayloadType,
+                clockRate: localCodec.clockRate,
+                numChannels: localCodec.numChannels || 1,
+                rtcpFeedback: [],
+                parameters: {}
+            };
+
+            /* eslint-disable max-len */
+            for (const remoteParamName of Object.keys(remoteCodec.parameters)) {
+                const remoteParamValue
+                    = remoteCodec.parameters[remoteParamName];
+
+                for (const localParamName of Object.keys(localCodec.parameters)) {
+                    const localParamValue
+                        = localCodec.parameters[localParamName];
+
+                    /* eslint-disable max-depth */
+                    if (localParamName !== remoteParamName) {
+                        continue; // eslint-disable-line no-continue
+                    }
+
+                    // TODO: We should consider much more cases here, but Edge
+                    // does not support many codec parameters.
+                    if (localParamValue === remoteParamValue) {
+                        // Use this RTP parameter.
+                        codec.parameters[localParamName] = localParamValue;
+                        break;
```

Comment:
No, we just found the mathing param so let's break from the inner `for`. Main `for` stills works.